### DISCUSSION TITLE
How to handle communication between two Microservices via EDC?

### DISCUSSION THREAD

**Discussion by SchattenPirat** [eclipse-edc/Connector#1441] [Q&A] [answered]
Hi,
I am interested in trying out the Eclipse Data Space Connector (EDC) for a certain scenario but I am not sure about how to realize it. Maybe you can help.
So far I was able to successfully execute this sample on my local machine [https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/tree/main/samples/04.0-file-transfer](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/tree/main/samples/04.0-file-transfer)
 
In my scenario, I have two EDCs and two microservices. The data flow should happen between the two microservices via the EDC connectors. Let's call the two sides that participate in the data flow
the “provider side” and the “consumer side”. Both sides consist of one microservice and one EDC connector. The microservice of the consumer side calls the transfer API of the consumer EDC.
Then the Consumer EDC triggers the communication with the Provider EDC. The supplier EDC doesn’t provide the “provider side” just a static file (as it is done in the sample) instead the provider EDC should fetch
the requested data via API calls to the provider microservice. To keep it simple Authentication is out of scope in this first Proof of Concept.
 
Now, I wonder what the best practice could be to let the “provider EDC” communicate with the “provider service” via API calls and how to send the fetched data to the “consumer EDC”.
I would appreciate suggestions.

kind regards
Bernd

**Comment by jimmarino** [comment, marked as answer]
You can either implement an HTTP data plane endpoint, a DataPlaneExtension framework extension or create a DataFlowControllerExtension. This is covered in the onboarding documentation.

**Reply by florianrusch-zf** [reply]
@Bernd-Siemer As I also wrote in another [discussion](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/discussions/1578), you could have a look at this extensions which is something like a proxy-flow through the EDC: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/tree/main/extensions/data-plane-transfer/data-plane-transfer-sync
